sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -0,0 +1,197 @@
+                        if (lowestBlockIndex === null || blocks[blockIndex].Level < blocks[lowestBlockIndex].Level) {
+                            lowestBlockIndex = blockIndex;
+                        }
+                    }
+
+                    Ext.Array.splice(blocks, lowestBlockIndex, 1);
+                }
+                
+                // sort counted demonstrations first
+                Ext.Array.sort(blocks, function(a, b) {
+                    if (a.counted && !b.counted) {
+                        return -1;
+                    }
+
+                    if (!a.counted && b.counted) {
+                        return 1;
+                    }
+
+                    return a.DemonstrationID > b.DemonstrationID ? 1 : -1;
+                });
+
+
+                // add empty blocks
+                while (blocks.length < demonstrationsRequired) {
+                    blocks.push({});
+                }
+                
+                if(blocks[0] == {}) {
+                    debugger;
+                }
+       
+                return blocks;
+            }
+        }
+    ],
+    
+    html: [
+      '<ul class="cbl-competency-panels competencies-unloaded" id="studentDashboardCompetenciesList"></ul>'  
+    ],
+
+    listeners: {
+        scope: 'this',
+        click: {
+            fn: 'onListClick',
+            element: 'el',
+            delegate: '.cbl-competency-panels, .cbl-competency-panels'
+/*
+        },
+        mouseover: {
+            fn: 'onSkillNameMouseOver',
+            element: 'el'
+*/
+        }
+    },
+
+    applyPopover: function(newPopover, oldPopover) {
+        return Ext.factory(newPopover, 'Slate.cbl.widget.Popover', oldPopover);
+    },
+
+    applyContentArea: function(contentArea) {
+        if (!contentArea) {
+            return null;
+        }
+
+        if (contentArea.isModel) {
+            return contentArea;
+        }
+
+        if (contentArea === true) {
+            contentArea = {};
+        }
+
+        return Ext.create('Slate.cbl.model.ContentArea', contentArea);
+    },
+
+    updateContentArea: function(newContentArea, oldContentArea) {
+        this.fireEvent('contentareachange', this, newContentArea, oldContentArea);
+    },
+
+    onListClick: function(ev, t) {
+        var me = this,
+            targetEl;
+
+        if (targetEl = ev.getTarget('.cbl-skill-demo', me.el, true)) {
+            me.fireEvent('democellclick', me, ev, targetEl);
+        }
+    },
+
+    onSkillNameMouseOver: function(ev) {
+        var me = this,
+            popover = me.getPopover(),
+            dashboardEl = me.el,
+            targetEl;
+
+        if (targetEl = ev.getTarget('.cbl-skill-name', dashboardEl, true)) {
+            if (popover.hidden || popover.alignTarget !== targetEl) {
+                popover.showBy(targetEl, 'tl-bl', [0, 42]);
+                popover.setWidth(targetEl.getWidth());
```

Comment:
Yes, but I believe this whole `onSkillNameMouseOver` method is deprecated and should be deleted as @ryon found a CSS-only solution to showing details on hover (and was probably incited to by this exact issue)